src/sql/workbench/services/notebook/browser/models/notebookModel.ts

```diff
@@ -573,6 +573,7 @@ export class NotebookModel extends Disposable implements INotebookModel {
 				let newCell = undefined, tailCell = undefined, partialSource = undefined;
 				let newCellIndex = index;
 				let tailCellIndex = index;
+				let splitCells: ICellModel[] = [];
 				let newLinesRemoved: string[] = [];
```

Comment:
This doesn't need to be an array, right? We know we're just removing a single string so let's do that to reduce possible confusion. And I'd also suggest updating the name to be more specific like `newlinesBeforeTailCellContent`

But also thinking about it - it would be even better to make the list of splitCells be a list of objects like this : 

```
type SplitCell {
    cell: ICellModel;
    prefix: string | undefined;
}
...
let splitCells: SplitCell[] = [];
```

So that you can then tell each cell whether it has prefix content that was stripped off or not (which would fix the issue I called out below with adding the newlines for every cell)